Content/ContentTypeResolver/ImageMapResolver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver;
+
+use Sulu\Bundle\HeadlessBundle\Content\ContentResolverInterface;
+use Sulu\Bundle\HeadlessBundle\Content\ContentView;
+use Sulu\Bundle\HeadlessBundle\Content\Serializer\MediaSerializerInterface;
+use Sulu\Bundle\MediaBundle\Media\Manager\MediaManagerInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class ImageMapResolver implements ContentTypeResolverInterface
+{
+    /**
+     * @var MediaManagerInterface
+     */
+    private $mediaManager;
+
+    /**
+     * @var MediaSerializerInterface
+     */
+    private $mediaSerializer;
+
+    /**
+     * @var ContentResolverInterface
+     */
+    private $resolver;
```

Comment:
would call this `$contentResolver` that it is more clear which resolver here is  used